backend/src/impl/default_controllers_impl.py

```diff
@@ -195,6 +195,9 @@ def systems_system_id_outputs_get(
     TODO: return special error/warning if some ids cannot be found
     """
     sys = SystemDBUtils.find_system_by_id(system_id)
+    user = get_user()
+    if not user.is_authenticated or (sys.is_private and sys.creator != user.email):
```

Comment:
I think this would require users to log in before viewing outputs for public systems. Is it by design?

Reply:
Fixed.